src/exporting/prometheus/prometheus.c

```diff
@@ -602,6 +597,240 @@ static void prometheus_print_os_info(
+                char *suffix = "";
+
+                struct gen_parameters p;
+                p.prefix = prefix;
+                p.labels_prefix = plabels_prefix;
+                p.context = context;
+                p.suffix = suffix;
+                p.chart = chart;
+                p.dimension = dimension;
+                p.family = family;
+                p.labels = (char *)opts->labels;
+                p.output_options = output_options;
+                p.st = st;
+                p.rd = rd;
+
+                if (as_collected) {
+                    // we need as-collected / raw data
+
+                    if (unlikely(rd->collector.last_collected_time.tv_sec < opts->instance->after))
+                        continue;
+
+                    p.type = "gauge";
+                    p.relation = "gives";
+                    if (rd->algorithm == RRD_ALGORITHM_INCREMENTAL ||
+                        rd->algorithm == RRD_ALGORITHM_PCENT_OVER_DIFF_TOTAL) {
+                        p.type = "counter";
+                        p.relation = "delta gives";
+                        if (!prometheus_collector)
+                            p.suffix = "_total";
+                    }
+
+                    if (plot_help) {
+                        generate_as_collected_prom_help(wb, prefix, context, units, suffix, st);
+                        plot_help = false;
+                        opts->output_options &= ~PROMETHEUS_OUTPUT_HELP;
+                    }
+
+                    if (plot_type) {
+                        generate_as_collected_prom_type(wb, prefix, context, units, p.suffix, p.type);
+                        plot_type = false;
+                        opts->output_options &= ~PROMETHEUS_OUTPUT_TYPES;
+                    }
+
+                    if (homogeneous) {
+                        // all the dimensions of the chart, has the same algorithm, multiplier and divisor
+                        // we add all dimensions as labels
+
+                        prometheus_label_copy(
+                            dimension,
+                            (output_options & PROMETHEUS_OUTPUT_NAMES && rd->name) ? rrddim_name(rd) : rrddim_id(rd),
+                            PROMETHEUS_ELEMENT_MAX);
+
+                        generate_as_collected_from_metric(wb, &p, homogeneous, prometheus_collector, st->rrdlabels);
+                    }
+                    else {
+                        // the dimensions of the chart, do not have the same algorithm, multiplier or divisor
+                        // we create a metric per dimension
+
+                        prometheus_name_copy(
+                            dimension,
+                            (output_options & PROMETHEUS_OUTPUT_NAMES && rd->name) ? rrddim_name(rd) : rrddim_id(rd),
+                            PROMETHEUS_ELEMENT_MAX);
+
+                        generate_as_collected_from_metric(wb, &p, homogeneous, prometheus_collector, st->rrdlabels);
```

Comment:
This may be moved out of the if/else since it is called anyway?

Reply:
Algorithm was updated.